### PR TITLE
Update documentation for legion_prof.

### PR DESCRIPTION
**Pull request by syamajala**
Add documentation for the different modes of the rust profiler.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -77,27 +77,34 @@ OLCF, NERSC, and the Swiss National Supercomputing Centre (CSCS).
 <tr>
 <td>David Zhang</td>
 <td>Seshu Yamajala</td>
-<td>Wei Wu</td>
```
Comment (elliottslaughter) [inline index.md:80]: We do need to update this file, but let's split that into a different PR.

File: profiling.md [hunk 2]
```diff
@@ -53,62 +51,35 @@ Legion Prof outputs its logs in a compressed binary format using `ZLIB`.
 If you don't have `ZLIB` on your system, you can set `USE_ZLIB=0` in
 your Makefile.
 
-### Generating a Profile
+## Generating Log Files
```
Comment (elliottslaughter) [inline profiling.md:54]: Maybe `Generating Legion Prof Logs`?

File: profiling.md [hunk 3]
```diff
@@ -38,8 +38,6 @@ If using GASNet, ensure that [GASNet has been properly configured and
 optimized]({{ "/gasnet/" | relative_url }}). GASNet can be important for performance on
 multi-node runs.
 
-## Legion Prof
```
Comment (elliottslaughter) [inline profiling.md:41]: Please be sure to go to the top of this file to update the table of contents.

File: profiling.md [hunk 4]
```diff
@@ -53,62 +51,35 @@ Legion Prof outputs its logs in a compressed binary format using `ZLIB`.
...
-below for usage instructions for the Rust profilers.
+Before post processing the log files output from running your
+application with `-lg:prof` the post processer must be installed by
+running the following:
 
 {% highlight bash %}
 curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
-cargo install --locked --path legion/tools/legion_prof_rs
-legion_prof prof_*.gz
+cargo install --locked --features=viewer --features=server --features=archiver --features=client --path legion/tools/legion_prof_rs
```
Comment (elliottslaughter) [inline profiling.md:73]: I remember we talked about this, but I don't remember why we decided not to just say `--all-features`, which is substantially shorter.

File: profiling.md [hunk 5]
```diff
@@ -53,62 +51,35 @@ Legion Prof outputs its logs in a compressed binary format using `ZLIB`.
...
-binary into `~/.cargo/bin`. The third runs the profiler. The supported
-flags are substantially similar to the Python implementation and can
-be queried with `--help`.
+binary into `~/.cargo/bin`. The supported flags can be queried with
+`legion_prof --help`.
+The profiler has four different modes it can be installed with which
+support various features.
 
-The following features are supported:
+By default all modes support the following features:
```
Comment (elliottslaughter) [inline profiling.md:82]: Let's delete from here to the next section. This is obsolete since we are now at feature parity, and anyway a comparison with the legacy profiler doesn't matter now.

File: profiling.md [hunk 6]
```diff
@@ -121,6 +92,88 @@ The following features are NOT currently supported:
   * Critical path analysis
   * Task dependencies
 
+### Legacy Profile
```
Comment (elliottslaughter) [inline profiling.md:95]: Let's move this to the bottom of this section, since we want to encourage people to use the new modes.

Could name it: `Legacy Profile Viewer`

File: profiling.md [hunk 7]
```diff
@@ -121,6 +92,88 @@ The following features are NOT currently supported:
...
+
+![]({{ "/images/profiling/legion_prof_web_viewer.png" | relative_url }})
+
+### Serve and Attach
+
+{% highlight bash %}
+legion_prof --serve prof_*.gz
+{% endhighlight %}
+
+Using this mode log files can be processed remotely on machine with
```
Comment (elliottslaughter) [inline profiling.md:154]: `on a machine`

**Review by elliottslaughter — APPROVED** [review]
Looks good, just some minor cleanup, some parts that can be deleted, and splitting the active contributors list.

File: index.md (hunk 1 above)
Comment (syamajala) [reply index.md:80]: Reverted.

File: profiling.md (hunk 2 above)
Comment (syamajala) [reply profiling.md:54]: Done.

File: profiling.md (hunk 4 above)
Comment (syamajala) [reply profiling.md:73]: Done.

File: profiling.md (hunk 5 above)
Comment (syamajala) [reply profiling.md:82]: Done.

File: profiling.md (hunk 6 above)
Comment (syamajala) [reply profiling.md:95]: Should we update the default mode of the profiler? That was the reason I put this first because I assumed most people would run the new legion_prof like the old one by default, without passing any flags to it. I'm not really sure which mode makes sense to be the new default though? Maybe archive?

Comment (elliottslaughter) [reply profiling.md:95]: Yeah, this is something we should think about. I'm not sure what the right answer is. I've been trying to avoid breaking people's workflows, so I haven't tried to change the interface so far.

One thing I thought about was switching to subcommands, so instead of `legion_prof --view` you write `legion_prof view`. And if we're going to make a change like that, we could give a name to the legacy mode while we're at it (e.g., `legion_prof old_archive` or `legion_prof v1_archive` or whatever).

If we have a default mode, I think `view` makes the most sense? Usually you just want to see the logs as easily as possible, and `--view` is the one-step process to do that. Everything else requires multiple steps and some decision on where to put things.

File: profiling.md (hunk 7 above)
Comment (syamajala) [reply profiling.md:154]: Done

File: index.md (hunk 1 above)
Comment (elliottslaughter) [reply index.md:80]: I still see changes in this file in the diff view?

Comment (elliottslaughter) [reply index.md:80]: Did you forget to push the branch?

File: profiling.md (hunk 3 above)
Comment (syamajala) [reply profiling.md:41]: Done.

**Comment by syamajala** [comment]
I think I finished everything. I could not run jekyll to view it so we should probably do that before merging this pull request?

File: profiling.md [hunk 8]
```diff
@@ -9,12 +9,14 @@ necessary to profile and tune for performance. This page covers
 techniques to achieving high performance in Legion applications.
 
   * [Preliminary Steps](#preliminary-steps)
+  * [Generating Legion Prof Logs](#generating-legino-prof-logs)
```
Comment (elliottslaughter) [inline profiling.md:12]: Typo here in the link (characters swapped).

**Review by elliottslaughter — COMMENTED** [review]
Thanks, this looks almost ready. My main remaining thought is whether we should capture all the screenshots in light mode? (Since the Legion website itself is light mode, and to keep a uniform look.) What do you think?

File: index.md (hunk 1 above)
Comment (syamajala) [reply index.md]: The last commit reverted the file back to the original. Under files changed I no longer see index.md.

**Comment by syamajala** [comment]
> Thanks, this looks almost ready. My main remaining thought is whether we should capture all the screenshots in light mode? (Since the Legion website itself is light mode, and to keep a uniform look.) What do you think?

I redid the image in light mode. I also just temporarily inverted the colors of my terminal since otherwise half the image would be light and half would be dark. 

Can we preview the changes before merging?

**Comment by syamajala** [comment]
Was there still more to do on this? If not I don't know how to deploy it myself.
